packages/keyring-snap-client/src/RestrictedKeyringClient.ts

```diff
@@ -0,0 +1,47 @@
+import { KeyringRpcMethod } from '@metamask/keyring-api';
+
+import type { KeyringClient } from './KeyringClient';
+
+/**
+ * Convert a tuple to a union.
+ */
+type TupleToUnion<Tuple extends readonly unknown[]> = Tuple[number];
+
+/**
+ * Extract the method name from a RPC method name.
+ */
+type KeyringRpcMethodToMethodName<RpcMethod extends string> =
+  RpcMethod extends `keyring_${infer Method}` ? Method : never;
+
+/**
+ * Restricted methods list.
+ */
+export const RestrictedKeyringRpcMethod = [
```

Comment:
I think **restricted** is not the best name here, it gives the impression that it's something that requires more rights, which is not completely correct. Maybe `Exposed*`, `External*`, `Public*`, or even `CompanionDapp*`.

Reply:
Yes, that's why I wanted your opinion on this 😄

I also thought of `Public*` TBH, but `CompanionDapp*` is also a good candidate 🤔 basically I wanted to expose that somewhere on the `accounts`-packages, so we can re-use it afterward on the extension/mobile-app.